platforms/core-configuration/model-core/src/main/java/org/gradle/api/internal/provider/AbstractCollectionProperty.java

```diff
@@ -574,55 +689,40 @@ private ExecutionTimeValue<C> getFixedExecutionTimeValue(List<ExecutionTimeValue
 
         @Override
         public ValueProducer getProducer() {
-            return value.getProducer();
-        }
-
-        @Override
-        public String toString() {
-            return value.toString();
-        }
-    }
-
-    /**
-     * A provider for a collection type whose elements are themselves providers.
-     */
-    private static class CollectingProvider<T, C extends Collection<? extends T>> extends AbstractMinimalProvider<C> {
-        private final Class<C> type;
-        private final List<ProviderInternal<? extends Iterable<? extends T>>> providers;
-        private final Supplier<ImmutableCollection.Builder<T>> collectionFactory;
+            List<ValueProducer> producers = getCollectors().stream().map(ValueSupplier::getProducer).collect(toList());
```

Comment:
❓ Does it actually need to collect here? From a behavior perspective, collecting here definitely means the returned `ValueProducer` won't reflect newly added elements. That should be sufficient for now but I wonder if it might not be surprising in the future. From a memory consumption perspective it would probably be nicer if the operations below could be Stream-based.

Reply:
> ❓ Does it actually need to collect here? From a behavior perspective, collecting here definitely means the returned `ValueProducer` won't reflect newly added elements. That should be sufficient for now but I wonder if it might not be surprising in the future. From a memory consumption perspective it would probably be nicer if the operations below could be Stream-based.

Makes sense, fixed.